interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is a user library within JavaScript. This can be used to created apps, without starting from scratch every single time, it basically has the bare bones of a structure of which you then build your app around. One of the syntax difference between HTML and JSX is how you assign a class to an element, in HTML it is simply class= "" in JSX it is className= ""
 
-Researched answer:
+Researched answer: JSX is a syntax extention of JavaScript. With JSX we are able to embed JavaScript objects within an HTML element.
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-Your answer:
+Your answer: Yarn is a function that we can use to test our code to see if it will work as it should. Also, yarn can be used to create an app that we can run on our local server so we can see the app and how it changes as we begin to write our code.
 
-Researched answer:
+Researched answer: Yarn is a package manger that installs a piece of code from a global registry onto the local computer. Yarn was designed to have clean output logs for easy understandability. Yarn stores dependencies locally. Yarn generates a yarn.lock file. 
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: An anonymous function is when you need a function to run but you don't give put anything in it. This is done by using an empty (). 
 
-Researched answer:
+Researched answer: An anonymous function is when a function has been declared, however the function doesn't have a name. Anonymous functions can have multiple arguments, but only one expression. It is not accessable after its initial creation.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are consistent, they don't change. Props aka (properties) can change. For example, if you have a cup of water, the state of that substance will always be water. However, you can change the propety of that water if you freeze it it will become ice, and if you boil it it will become steam... but the state of that substance will always remain water.
 
-Researched answer:
+Researched answer: The state is controlled by the component, while props are controlled by whatever renders the comonent. Props will get data from an external source. States manage the internal data.
```

Comment:
State is a special variable in React and the component is actually controlled by state in a sense.